### ISSUE TITLE
Gradle task consumes AGP IdentityTransform output without declaring it as a task input (Gradle 9.7 deprecation)

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#9742]
## Expected Behavior

Running detekt on an Android module that depends on a plain Kotlin/JVM project should not emit deprecation warnings on Gradle 9.7+. The artifact produced for the project dependency (AGP's `IdentityTransform`) should be declared as a task input.

## Observed Behavior

On Gradle 9.7.1, `:android-lib:detektDebug` emits:

```
Querying the output of an artifact transform from a task action without declaring it as a task input has been deprecated. This is scheduled to be removed in Gradle 10. Declare the files or artifacts produced by the configuration using the transform as a task input to properly wire it into the execution plan. Consult the upgrading guide for further information: https://docs.gradle.org/9.7.1/userguide/upgrading_version_9.html#undeclared_artifact_transform_input
	at org.gradle.api.internal.artifacts.transform.TransformStepNode.nagAboutUndeclaredResolution(TransformStepNode.java:281)
	at org.gradle.api.internal.artifacts.transform.TransformStepNode.executeIfNotAlready(TransformStepNode.java:254)
	at org.gradle.api.internal.artifacts.transform.TransformedProjectArtifactSet.visit(TransformedProjectArtifactSet.java:87)
	...
```

The same build on Gradle 9.6.1 completes without this warning. With `org.gradle.warning.mode=fail` it becomes a hard build failure on Gradle >= 9.7.0.

## Steps to Reproduce

Minimal repro: <https://github.com/TWiStErRob/repros/tree/main/detekt/undeclaredArtifactTransform-gradle97>

- `jvm-lib/`: plain JVM Kotlin library (`org.jetbrains.kotlin.jvm` + `dev.detekt`)
- `android-lib/`: Android library (`com.android.library` + `dev.detekt`) with `dependencies { implementation(project(":jvm-lib")) }`

Run:

```bat
gradlew :android-lib:detektDebug --rerun-tasks --warning-mode all
```

- Gradle 9.7.1: build succeeds, deprecation warning printed above.
- Gradle 9.6.1 (change `distributionUrl` in `gradle/wrapper/gradle-wrapper.properties`): no warning at all.

## Context

Fixing this is required for Gradle 10 compatibility and already breaks CI with `org.gradle.warning.mode=fail` on Gradle 9.7.x.

## Your Environment

* Version of detekt used: 2.0.0-alpha.6 (`dev.detekt` Gradle plugin)
* Version of Gradle used (if applicable): 9.7.1 (fails) / 9.6.1 (works)
* Gradle scan link (add `--scan` option when running the gradle task): not provided
* Operating System and version: Windows 11, JDK 21 (target compatibility 17), AGP 9.4.1, Kotlin 2.4.20
* Link to your project (if it's a public repository): <https://github.com/TWiStErRob/repros/tree/main/detekt/undeclaredArtifactTransform-gradle97>
